Allow strings py3

kelvinatorr opened the pull request:
This allows users to pass in strings for keys and values in python3 which makes the interface similar to how it was in python2. I also added a bunch more unittests.

Commit:
Add allowing python strings for keys and values

Commit:
Add returning py strings if the value read from the mtbl is decodeable

Commit:
Allow python strings for reader accessors and add a lot more unittests

Commit:
Fix merger unittest and add a test where merging actually happens

Commit:
Fix search unittest

Commit:
Allow str in varint_decode

This lets users use varint_decode() to decode varint_encoded integers
0-127 stored in an mtbl in py3

Commit:
Add more merger tests with varints

Commit:
Fix Readme 1st example is back to py2 style and add additional info

Commit:
Add a test for sorting

Commit:
Fix deprecation warning for locale.format()

hstern commented on mtbl.pyx:
```diff
@@ -120,7 +148,7 @@ cdef class iterkeys(object):
         if res == mtbl_res_failure:
             raise StopIteration
-        return <bytes> key[:len_key]
+        return from_bytes(key[:len_key])
```
I feel a bit funny with this being the default behaviour with no way to disable it. How about we add a kwargs bool to __init__ to turn it off?

kelvinatorr replied:
Okay I added a "return_bytes" option when creating a reader, merger, or sorter. The default is False so if you pass in 'abc' you will get back 'abc' like you did in Python2. But if you initialize your reader `return_bytes=True` you'll get back b'abc'

I added a bunch of tests for that functionality as well.

hstern reviewed (commented):
Love all of the tests.

Commit:
Add venv wildcard to gitignore

This will cover venvpy2 or venvpy3 for py2 and py3 virtual
environments for example

Commit:
Add example of varint encoding and decoding of large int

Commit:
Add kwargs init option to return bytes for reader, merger, and sorter

Commit:
Add stuct pack and unpack example to readme

kelvinatorr commented on README.md:
```diff
@@ -9,45 +9,80 @@ writer interfaces (nb. the traceback is expected behavior):
+r = mtbl.reader('example.mtbl', verify_checksums=True)
+
+assert mtbl.varint_decode(r.get('key0')[0]) == 2
+assert mtbl.varint_decode(r.get('key1')[0]) == 128
```
It actually works. I tried:

assert 987654321000 == mtbl.varint_decode(mtbl.varint_encode(987654321000))

and it worked. I added it as an example so it is clear.

hstern replied:
Is `r.get('key0')[0]` not taking the first byte of the value of `r.get('key0')` which would truncate a larger varint?

kelvinatorr replied:
No, reader.get() actually returns a list. So in the 128 example `r.get('key1')` returns `[b'\x80\x01']`. The [0] is just to access the first element for the comparison.

kelvinatorr commented on README.md:
````diff
@@ -9,45 +9,80 @@ writer interfaces (nb. the traceback is expected behavior):
+assert r.get('key3') == [b'\x80\x01'] # not a printable string so the value read back is bytes
+```
+
+If you want to store integers in your mtbl use varint_encode() and varint_decode() to do so:
````
Yep; I added that to the readme
